### PR TITLE
add delete files in range with delete range test

### PR DESCRIPTION
**Pull request by huachaohuang**
This test will not pass until https://github.com/facebook/rocksdb/pull/2843 is merged.

### CODE REVIEW

File: tests/test_delete_files_in_range.rs [hunk 1]
```diff
@@ -128,3 +129,71 @@ fn test_delete_files_in_range_with_snap() {
...
+        let mut v = vec![0; value_size];
+        rand::thread_rng().fill_bytes(&mut v);
+        db.put(k1.as_bytes(), v.as_slice()).unwrap();
+        db.put(k2.as_bytes(), v.as_slice()).unwrap();
+        db.flush(true).unwrap();
+    }
+
+    // After this, the delete range [0, 6) will drop all entries
+    // before it, so we should have only keys 4 and 5.
+    db.compact_range(None, None);
```
Comment (ajkr) [inline tests/test_delete_files_in_range.rs]: Is this where it fails with an assertion failure? If not, would you mind please posting the failure details?

**Comment by siddontang** [comment]
any update @huachaohuang

**Comment by UncP** [comment]
LGTM

**Pull request merged**